src/api/app/models/workflow/step.rb

```diff
@@ -95,6 +95,32 @@ def source_project_name
 
   private
 
+  def scm_synced?
+    scm_synced_project_url.present? || scm_synced_package_url.present?
+  end
+
+  def scmsync_url
+    return scm_synced_package_url if scm_synced_package?
+
+    scm_synced_project_url
+  end
+
+  def scm_synced_package?
+    scm_synced_package_url.present?
+  end
+
+  def scm_synced_project?
+    scm_synced_project_url.present?
+  end
+
+  def scm_synced_package_url
+    Package.get_by_project_and_name(source_project_name, source_package_name).try(:scmsync)
```

Comment:
What about [Links](https://github.com/openSUSE/open-build-service/wiki/Links)?

Reply:
Links are not exclusively about sources but in this case they don't really play a role. As long as you have a Package/Project object it might have an scmsync attribute and that is everything we are interested in.